src/scope.ml

```diff
@@ -120,6 +120,73 @@ type mode =
       carries the environment for variables that will be bound in the
       representation of the RHS. *)
 
+(** [get_params_implicitness_of_type t] returns a list representing the
+    implicitness of the parameters of a parser term [t] seen as a type *)
+    let get_params_implicitness_of_type : p_term -> bool list = fun t ->
+    match t.elt with
+    | P_Prod(args, _)   -> List.map (fun x -> match x with (_,_,i) -> i) args
+    | _                 -> []
+
+(** [get_params_implicitness_of_id ss env t] returns a list representing the
+    formal parameters of a parser term [t] *)
+let get_params_implicitness_of_id : sig_state -> env -> p_term -> bool list =
+  fun ss env t ->
+  match t.elt with
+  | P_Iden(_, FullyExplicit)       -> []
+  | P_Iden(id, ImplicitAsDeclared) ->
+      (* We first look in the environment *)
```

Comment:
All this code seems a duplication of find_sym and find_qid's job. Why not adding implicits *after* scoping? That is, doing as follows when doing the scoping of a P_Iden or P_Appl: 1. do splitting. 2. do scoping. 3. if the head is a symb then get its sym_implicits and add implicit arguments (scoping code of     | (P_Wild          , M_Term(_)) ->
that should be factorized).